src/shared/components/CSVExportButton.tsx

```diff
@@ -1,72 +1,102 @@
 // Libraries
 import React, {PureComponent} from 'react'
-import {connect} from 'react-redux'
+import {connect, ConnectedProps} from 'react-redux'
 
 // Components
-import {Button, ComponentStatus, IconFont} from '@influxdata/clockface'
-
-// Utils
-import {downloadTextFile} from 'src/shared/utils/download'
-import {getActiveTimeMachine} from 'src/timeMachine/selectors'
-import {createDateTimeFormatter} from 'src/utils/datetime/formatters'
+import {
+  Button,
+  ComponentStatus,
+  ComponentColor,
+  IconFont,
+} from '@influxdata/clockface'
+
+// Selectors and Actions
+import {getActiveQuery} from 'src/timeMachine/selectors'
+import {
+  runDownloadQuery,
+  DOWNLOAD_EVENT_COMPLETE,
+} from 'src/timeMachine/actions/queries'
 
 // Types
-import {AppState} from 'src/types'
+import {AppState, RemoteDataState} from 'src/types'
+
+type Props = ConnectedProps<typeof connector>
 
-interface StateProps {
-  files: string[] | null
+interface State {
+  state: RemoteDataState
 }
 
-class CSVExportButton extends PureComponent<StateProps, {}> {
+class CSVExportButton extends PureComponent<Props, State> {
+  private _controller: AbortController
+
+  constructor(props) {
+    super(props)
+    this._controller = new AbortController()
+    this._controller.signal.addEventListener(DOWNLOAD_EVENT_COMPLETE, () => {
+      this.setState({state: RemoteDataState.Done})
+    })
+    this.state = {state: RemoteDataState.NotStarted}
+  }
+
   public render() {
+    if (this.state.state == RemoteDataState.Loading) {
```

Comment:
Yeah, this is a good example of why I'm unsure wrapping a single object in a single object makes sense. `this.state.state` has a weird stutter, and has a "yo dawg" feel about state, e.g. "we heard you like state, so I gave your state some state, so you can check the state of your state."

Reply:
Renamed to `loadingState`, to avoid the stutter.